crates/bevy_ecs/src/term_query/terms/mod.rs

```diff
@@ -0,0 +1,363 @@
+use bevy_utils::all_tuples;
+
+use crate::{
+    archetype::{Archetype, ArchetypeComponentId},
+    component::{ComponentId, Tick},
+    entity::Entity,
+    prelude::World,
+    query::{Access, DebugCheckedUnwrap, FilteredAccess},
+    storage::{Table, TableRow},
+    world::unsafe_world_cell::UnsafeWorldCell,
+};
+
+mod component;
+mod entity;
+mod group;
+
+pub use component::*;
+pub use entity::*;
+pub use group::*;
+
+use super::TermVec;
+
+#[derive(Clone)]
+pub enum Term {
+    Entity(EntityTerm),
+    Component(ComponentTerm),
+    Or(OrTerm),
+}
+
+#[derive(Eq, PartialEq, Clone)]
+pub enum TermAccess {
```

Comment:
We should probably try to unify this with the existing access code. I'm not convinced it should be in this PR, but we *really* need to avoid divergences.